src/inputs/TreeSelectField/TreeSelectField.tsx

```diff
@@ -157,29 +164,53 @@ function TreeSelectFieldBase<O, V extends Value>(props: TreeSelectFieldProps<O,
 
   const { collapsedKeys } = useTreeSelectFieldProvider();
 
-  function levelOptions(o: NestedOption<O>, level: number, filtering?: boolean): LeveledOption<O>[] {
-    // If a user is filtering, then do not provide level to the options as the various paddings may look quite odd.
-    const actualLevel = filtering ? 0 : level;
-    return [
-      [o, actualLevel],
-      ...(o.children?.length && !collapsedKeys.includes(valueToKey(getOptionValue(o)))
-        ? o.children.flatMap((oc: NestedOption<O>) => levelOptions(oc, actualLevel + 1, filtering))
-        : []),
-    ];
-  }
+  const levelOptions = useCallback(
```

Comment:
Question: I know this name was already here, but is this basically `flattenOptions`? Or `deepFlattenOptions`? As a potentially better name...